### PR TITLE
Drop support for Python 3.6

### PR DESCRIPTION
**Pull request by thijstriemstra**

### CODE REVIEW

**Review by rm-hull — APPROVED** [review]

**Review by rm-hull — APPROVED** [review]

**Comment by thijstriemstra** [comment]
luma.emulator is a WIP

**Review by rm-hull — APPROVED** [review]

**Review by rm-hull — APPROVED** [review]

**Review by rm-hull — APPROVED** [review]

**Comment by thijstriemstra** [comment]
Running tox results in failing tests: it can't find the embedded images, but running pytest shows no problems. This makes me think there's a bug in tox. Not sure what to do next.

**Comment by thijstriemstra** [comment]
> This makes me think there's a bug in tox. Not sure what to do next.

Looks like it indeed was a bug in tox because I tried it again with the latest and it now seems to work, yay.

[Tests fail](https://github.com/rm-hull/luma.emulator/actions/runs/5744253673/job/15570222620#step:8:31) with 2 errors and I think the 2nd test error (test_gifanim) is related to the first:

```
 =================================== FAILURES ===================================
__________________________________ test_alpha __________________________________

    def test_alpha():
        w, h = 100, 50
        device = emulator(width=w, height=h, rotate=3, mode='RGB',
            transform='none', scale=6)
    
        im = Image.new('RGB', (w, h), (255, 0, 0))
        dr = ImageDraw.Draw(im)
>       dr.ellipse((w / 2, h / 2, 10, 10), fill='black', outline='blue')

tests/test_emulator.py:45: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <PIL.ImageDraw.ImageDraw object at 0x7ff6ce17b1e0>
xy = (50.0, 25.0, 10, 10), fill = -16777216, outline = 'blue', width = 1

    def ellipse(self, xy, fill=None, outline=None, width=1):
        """Draw an ellipse."""
        ink, fill = self._getink(outline, fill)
        if fill is not None:
>           self.draw.draw_ellipse(xy, fill, 1)
E           ValueError: x1 must be greater than or equal to x0
```

any idea what might've changed (in Pillow) @rm-hull?

**Comment by thijstriemstra** [comment]
> any idea what might've changed (in Pillow)

~~Looks like it's related to https://github.com/python-pillow/Pillow/pull/7284 and https://github.com/python-pillow/Pillow/issues/7279 and it will be fixed in Pillow 10.1.0.~~

I tested again with Pillow master branch but that didn't fix it. Downgrading to 9.5.0, 9.1.0 didn't help either so the issue's been there for a while?

**Comment by thijstriemstra** [comment]
Ok, I fixed the tests, and bumped version to 1.5.0. Could you force merge this @rm-hull? I can't cause it's stuck on an (obsolete) python 3.6 github actions build.

File: tests/reference/anim.gif
Comment (thijstriemstra) [inline tests/reference/anim.gif:1]: the green triangle looks much better, Pillow definitely evolved.

**Comment by rm-hull** [comment]
Nice one

**Comment by thijstriemstra** [comment]
I tagged 1.5.0 and it started publishing using github actions but failed (so I removed the tag again) with:

```
Notice: Attempting to perform trusted publishing exchange to retrieve a temporary short-lived API token for authentication against https://test.pypi.org/legacy/ due to __token__ username with no supplied password field
Error: Trusted publishing exchange failure: 
```

See https://github.com/rm-hull/luma.emulator/actions/runs/5753077008/job/15595490117#step:7:20

can you doublecheck if the pypi/repo permission settings for luma.emulator are correct, and similar to luma.oled where it does work @rm-hull? Looks like some token is missing..

**Pull request merged**